dss/events/handlers/notify_v2.py

```diff
@@ -77,8 +77,12 @@ def notify_or_queue(replica: Replica, subscription: dict, metadata_document: dic
             for key in bundles - tombstones:
                 sqsm.send(_format_sqs_message(replica, subscription, event_type, key), delay_seconds=0)
         else:
-            if not notify(subscription, metadata_document, key):
+            notify_status = notify(subscription, metadata_document, key)
+            if notify_status:
+                update_subscription_stats(subscription, True)
+            else:
                 sqsm.send(_format_sqs_message(replica, subscription, event_type, key), delay_seconds=15 * 60)
+                update_subscription_stats(subscription, False)
```

Comment:
Consider:
`update_subscription_stats(subscription, "failed")`
and
`update_subscription_stats(subscription, "successful")`

Also in my mind it might make sense to change the order that it updates the status before sending it off again?
```
update_subscription_stats(subscription, False)
qsm.send(_format_sqs_message(replica, subscription, event_type, key), delay_seconds=15 * 60)
```
Instead of:
```
qsm.send(_format_sqs_message(replica, subscription, event_type, key), delay_seconds=15 * 60)
update_subscription_stats(subscription, False)
```

Reply:
good call